src/stx/stl/trex_stl_dp_core.cpp

```diff
@@ -809,10 +809,12 @@ bool PerPortProfile::stop_traffic(uint8_t  port_id,
 
     for (auto& dp_stream : m_active_nodes) {
         CGenNodeStateless * node =dp_stream.m_node;
-        assert(node->get_port_id() == port_id);
```

Comment:
I don't like this change.  In this very file there are 7 assertions like this one.  Adding `if`s in one place does not look like a solution to me.

Reply:
Please let me know when you are finished witch changing `CGenNodeTimesync` structure.  I can help with aligning those attributes to match what is expected of `CGenNodeStateless`.

Reply:
As we discussed it previously aligning struct members is not a solution for this problem (As it is not standard layout/trivial data).
We should not relay on UB (as my test showed this can work for release but is failing when checked in UT) and do this as here or do this in right way.

If we want to do this right we need to move port_id (and probably other common members to base class).

Reply:
I do not like the idea of slowing down this “wildly optimized beast” (meaning TRex), that is why I was reluctant to adding more `if`s.  That said, I see `PerPortProfile::stop_traffic` and other methods declared in `trex_stl_dp_core.cpp` are not performance-critical.  So I guess I could live with the change you suggest.

There are six other occurrences of assertion like this one in `trex_stl_dp_core.cpp`, could you change them as well?  Did you check other classes/files for usage of `get_port_id()` that might need to be changed?

Reply:
About that “right way”…  I would certainly love to do some refactoring to get things right.  But I am not sure if it is the best use of our time and resources.  Especially if our fork is not intended to be merged with Cisco's…